src/classes/ElevateTokenizedGift.cls

```diff
@@ -0,0 +1,65 @@
+public virtual with sharing class ElevateTokenizedGift {
+    protected Decimal amount;
+    protected String firstName;
+    protected String paymentMethodToken;
+    protected String merchantId;
+    protected String gatewayId;
+    protected String lastName;
+    protected String currencyCode;
+    protected PS_IntegrationServiceConfig.Service configService;
+
+    public ElevateTokenizedGift() {
+        this.currencyCode = UserInfo.getDefaultCurrency();
+        setConfigServiceInstance();
```

Comment:
What do you think about inlining these setters? This constructor exists to support testing right?